pkg/hubble/observer/utils.go

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package observer
+
+import (
+	"fmt"
+	"strings"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+)
+
+type filter map[string]filter
+
+// createFilter constructs a tree filter based on validated and normalized field
+// mask fm. Use active() to check if applying a filter will have any effect.
+func createFilter(fm *fieldmaskpb.FieldMask) (filter, error) {
+	if fm == nil {
+		return nil, nil
+	}
+	if !fm.IsValid(&flowpb.Flow{}) {
+		return nil, fmt.Errorf("invalid fieldmask")
+	}
+	fm.Normalize()
+
+	f := make(filter)
+	for _, path := range fm.GetPaths() {
```

Comment:
should we be checking for empty paths?

Reply:
Do you mean when there is an empty slice of paths or if the path is empty (i.e. `""`)?

The former will produce a filter of length 0, so it will be "inactive" (no filter will be applied). The latter will return an error. Do you want to change any of these behaviors?

Reply:
I meant the latter case, I can't see where an error would be returned though.

Reply:
In L24, I added a test for that